identity/src/main/java/com/stripe/android/identity/ui/IDNumberSection.kt

```diff
@@ -0,0 +1,321 @@
+package com.stripe.android.identity.ui
+
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.navigation.NavController
+import com.stripe.android.core.model.Country
+import com.stripe.android.identity.R
+import com.stripe.android.identity.navigation.CountryNotListedDestination
+import com.stripe.android.identity.navigation.navigateTo
+import com.stripe.android.identity.networking.Resource
+import com.stripe.android.identity.networking.models.IdNumberParam
+import com.stripe.android.uicore.elements.CountryConfig
+import com.stripe.android.uicore.elements.CountryElement
+import com.stripe.android.uicore.elements.DropdownFieldController
+import com.stripe.android.uicore.elements.FieldError
+import com.stripe.android.uicore.elements.IdentifierSpec
+import com.stripe.android.uicore.elements.SectionElement
+import com.stripe.android.uicore.elements.SectionElementUI
+import com.stripe.android.uicore.elements.SimpleTextElement
+import com.stripe.android.uicore.elements.SimpleTextFieldConfig
+import com.stripe.android.uicore.elements.SimpleTextFieldController
+import com.stripe.android.uicore.elements.TextFieldState
+
+/**
+ * Section to collect User's ID number from different countries.
+ * Only US, BR and SG are supported.
+ */
+@Composable
+internal fun IDNumberSection(
```

Comment:
We're adding a bunch of items to the detekt baseline with this component. Could we clean it up a bit, extract stuff to methods, etc to avoid it?

Reply:
updated most of them, but still got 2 compose functions since we need to build some states to make SectionElement work, updated the baseline